pkg/client/contract.go

```diff
@@ -108,6 +126,32 @@ func (contract *Contract) Submit(transactionName string, options ...ProposalOpti
 	return result, nil
 }
 
+// SubmitWithContext submit a transaction to the ledger in the scope of a specific Context and return its result only after it has been committed to the ledger. This method
+// provides greater control over the transaction proposal content and the endorsing peers on which it is evaluated. This
+// allows transaction functions to be submitted where the proposal must include transient data, or that will access
+// ledger data with key-based endorsement policies.
+//
+// This method may return different error types depending on the point in the transaction invocation that a failure
+// occurs. The error can be inspected with errors.Is or errors.As.
+func (contract *Contract) SubmitWithContext(ctx context.Context, transactionName string, options ...ProposalOption) ([]byte, error) {
+
+	result, commit, err := contract.SubmitAsyncWithContext(ctx, transactionName, options...)
+	if err != nil {
+		return result, err
+	}
+
+	status, err := commit.StatusWithContext(ctx)
+	if err != nil {
+		return result, err
+	}
+
+	if !status.Successful {
+		return nil, fmt.Errorf("transaction %s rejected with validation code %d (%s)", status.TransactionID, int32(status.Code), peer.TxValidationCode_name[int32(status.Code)])
```

Comment:
This needs to return a CommitError, and there is an existing constructor function to create one.

```suggestion
		return nil, newCommitError(status.TransactionID, status.Code)
```

There should probably be a unit test to confirm that a CommitError is returned, like the one for Submit():

https://github.com/hyperledger/fabric-gateway/blob/68a5a1a875c874272fc3e280c61808f1145fdce0/pkg/client/submit_test.go#L159-L178

You might be able to convert some of the tests that are common between Submit() and SubmitWithContext() into parameterized tests. Whatever is easier.